src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlLoader.cs

```diff
@@ -85,7 +85,7 @@ private void LoadDocSequence(XmlDocument parentDoc)
                 XmlNode n = LoadNode(true)!;
 
                 // Move to the next node
-                if (n.NodeType != XmlNodeType.Attribute)
+                if (n != null && n.NodeType != XmlNodeType.Attribute)
```

Comment:
The null-forgiving operator in the previous line explicitly expressed that null shouldn't be returned in this case.

Reading through the implementation, null can only be returned in case of malformed xml (end element without starting element). In this case, the behavior should be reconsidered.

Reply:
It may be more appropriate to fix it like this then?
```
XmlNode? n = LoadNode(true);
Debug.Assert(n != null);
```